web/src/main/java/org/wso2/testgrid/web/api/SSOService.java

```diff
@@ -0,0 +1,153 @@
+
+    @Context
+    HttpServletRequest request;
+
+    /**
+      * This has the implementation of the REST API for creating session and send the redirection information .
+      * @return Redirection to dashboard home page.
+     */
+    @POST
+    public Response createSession(@FormParam("SAMLResponse") String responseMessage) throws TestGridException {
+        XMLObject response;
+        org.opensaml.saml2.core.Response saml2Response;
+        try {
+            response = SSOAgentUtils.unmarshall(new String(Base64.decode(responseMessage), Charset.forName("UTF-8")));
+            saml2Response = (org.opensaml.saml2.core.Response) response;
+        } catch (SSOAgentException e) {
+            String msg = "Error occurred while unmarshalling SAMLResponse.";
+            logger.error(msg, e);
+            return Response.serverError().entity(
+                    new ErrorResponse.ErrorResponseBuilder().setMessage(msg + e.getMessage()).build()).build();
+        }
+
+        try {
+            validateSignature(saml2Response.getSignature());
+            request.getSession();
+            return Response.status(Response.Status.FOUND).
+                    header(HttpHeaders.LOCATION, Constants.WEBAPP_CONTEXT).
+                    type(MediaType.TEXT_PLAIN).build();
+        } catch (SSOAgentException e) {
+            String msg = "Observed an attempt with invalid SAMLResponse.";
+            logger.error(msg, e);
+            return Response.status(Response.Status.UNAUTHORIZED).entity(msg + e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Validates the XML Signature object
+     *
+     * @param signature XMLObject
+     * @throws TestGridException if either SignatureProfile or Signature validation is failed.
+     */
+    private void validateSignature(XMLObject signature) throws SSOAgentException, TestGridException {
+
+        SignatureImpl signImpl = (SignatureImpl) signature;
+        try {
+            SAMLSignatureProfileValidator signatureProfileValidator = new SAMLSignatureProfileValidator();
+            signatureProfileValidator.validate(signImpl);
+        } catch (ValidationException e) {
+            throw new TestGridException("Signature do not confirm to SAML signature profile. Possible XML Signature " +
+                    "Wrapping Attack!", e);
+        }
+
+        try {
+            SignatureValidator validator = new SignatureValidator(
+                    new X509CredentialImpl(getX509Credential()));
+            validator.validate(signImpl);
+        } catch (ValidationException e) {
+            throw new TestGridException("Signature validation failed for SAML2 Element.");
+        }
+    }
+
+    /**
+     * Read JKS file and return X509Credential of Identity Server.
+     * @return X509Credential of Identity Server.
+     * @throws TestGridException if an error occur while reading JKS file.
+     */
+    private SSOAgentX509Credential getX509Credential() throws TestGridException {
```

Comment:
You are returning the identity provider's X509Credential certificate here, right? Method name does not capture that.